sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -583,6 +689,10 @@ class CudfRegexTranspiler(mode: RegexMode) {
               // Java, it's treated as a single (b$ and b$$ are synonymous), so we create 
               // an empty RegexAST that outputs to empty string
               RegexEmpty()
+            case Some(RegexChar(ch)) if mode == RegexReplaceMode
+                && lineTerminatorChars.contains(ch) =>
+                throw new RegexUnsupportedException("Regex sequences with a line terminator " 
```

Comment:
Can we add a fallback test to verify that this is working as expected?  Does this conflict with the test just below it? What happens if we have something like `ABC$A\n` is this going to do the right thing?

Reply:
I added a test for this in the follow-up PR https://github.com/NVIDIA/spark-rapids/pull/5519

This case statement is guarded on the condition `mode == RegexReplaceMode` and the one below it is not so there is no conflict between the two. The `ch` being tested here is prior to the `$` so would not match the `ABC$A\n` case.